pallets/gear/src/lib.rs

```diff
@@ -1600,9 +1600,11 @@ pub mod pallet {
             // overlay and never be committed to storage.
             GearRunInBlock::<T>::set(Some(()));
 
-            let weight_used = <frame_system::Pallet<T>>::block_weight();
+            // Subtract current block weight for Normal dispatches from the maximum block weight.
             let max_weight = <T as frame_system::Config>::BlockWeights::get().max_block;
-            let remaining_weight = max_weight.saturating_sub(weight_used.total());
+            let remaining_weight = max_weight.saturating_sub(
```

Comment:
How about the inherents (like `Timestamp::set()`) which also use some weight (very little though - ~10,500,000, but still) and have `DispatchClass::Mandatory`?
Also, we might have `DispatchClass::Operational` extrinsics one day. Should we account for those, as well, when calculating the remaining weight?